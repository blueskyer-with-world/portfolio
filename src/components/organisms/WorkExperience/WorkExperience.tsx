import { ContentBlock } from '@components/atoms/ContentBlock';
import { Job } from '@components/molecules/Job';

import { Job as JobType } from '@types';
import classNames from 'classnames';

export interface WorkExperienceProps {
	jobs: JobType[];
}

const WorkExperience = ({ jobs }: WorkExperienceProps) => {
	return (
		<>
			{jobs.map(
				({ company, jobTitle, fromDate, toDate, skills, description }) => (
					<div
						className={classNames(
							'flex transition-all transform md:hover:scale-[1.01]'
						)}
						key={`${company} ${jobTitle}`}
					>
						<Job
							company={company}
							jobTitle={jobTitle}
							fromDate={fromDate}
							toDate={toDate}
							skills={skills}
						>
							<ContentBlock value={description} />
						</Job>
					</div>
				)
			)}
		</>
	);
};

export { WorkExperience };
