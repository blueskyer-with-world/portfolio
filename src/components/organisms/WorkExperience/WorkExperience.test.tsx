import { WorkExperience } from '../WorkExperience';

import { mockJobs } from '@mockdata';
import { render } from '@testing-library/react';

describe('WorkExperience', () => {
	it('renders correctly', () => {
		const { container } = render(<WorkExperience jobs={[mockJobs[0]]} />);
		expect(container).toMatchSnapshot();
	});
});
