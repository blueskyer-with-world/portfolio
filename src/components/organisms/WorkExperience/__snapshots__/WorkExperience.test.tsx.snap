// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkExperience renders correctly 1`] = `
<div>
  <div
    class="flex transition-all transform md:hover:scale-[1.01]"
  >
    <div
      class="md:w-10/12 px-8 py-6 rounded-3xl glass-light dark:glass-dark mb-6 drop-shadow-lg hover:drop-shadow-2xl transition-all"
    >
      <div
        class="flex gap-4 items-start md:items-center"
      >
        <img
          alt="Holland & Barrett"
          class="rounded-md"
          data-nimg="1"
          decoding="async"
          height="64"
          loading="lazy"
          src="/assets/companies/holland-and-barrett.svg"
          style="color: transparent;"
          width="64"
        />
        <div
          class="w-full mb-2"
        >
          <div
            class="flex flex-col-reverse md:flex-row justify-between"
          >
            <h3
              class="font-bold"
            >
              Senior Software Engineer
            </h3>
            <span
              class="text-sm md:text-base"
            >
              Aug 2020 â€” present
            </span>
          </div>
          <h4>
            <a
              class="underlined relative"
              href="https://www.hollandandbarrett.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Holland & Barrett
            </a>
          </h4>
          <p
            class="hidden md:block text-sm"
          >
            <strong>
              Technologies: 
            </strong>
            TypeScript, React, Next.js, Redux, GraphQL, REST API, Jest, Cypress
          </p>
        </div>
      </div>
      <p
        class="md:hidden text-sm"
      >
        <strong>
          Technologies: 
        </strong>
        TypeScript, React, Next.js, Redux, GraphQL, REST API, Jest, Cypress
      </p>
      <div
        id="contentBlock"
      />
    </div>
  </div>
</div>
`;
